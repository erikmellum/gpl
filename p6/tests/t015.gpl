string window_title = "p6 t015.gpl -- compare to t015.jpg";
int window_width = 600;
int window_height = 600;

rectangle r1(x = 300, y = 300, w = 10, h = 10,
             red = 0.5, green = 1.0, blue = 1.0);

rectangle r2(x = r1.x - r1.w/2, y = r1.y - r1.h/2, w = r1.w*2, h = r1.h*2,
             red = r1.red, green = r1.green*.9, blue = r1.blue*.8,
             rotation = 30);

rectangle r3(x = r2.x - r2.w/2, y = r2.y - r2.h/2, w = r2.w*2, h = r2.h*2,
             red = r2.red, green = r2.green*.9, blue = r2.blue*.8,
             rotation = r2.rotation*2);

rectangle r4(x = r3.x - r3.w/2, y = r3.y - r3.h/2, w = r3.w*2, h = r3.h*2,
             red = r3.red, green = r3.green*.9, blue = r3.blue*.8,
             rotation = r2.rotation*3);

rectangle r5(x = r4.x - r4.w/2, y = r4.y - r4.h/2, w = r4.w*2, h = r4.h*2,
             red = r4.red, green = r4.green*.9, blue = r4.blue*.8,
             rotation = r2.rotation*4);

rectangle r6(x = r5.x - r5.w/2, y = r5.y - r5.h/2, w = r5.w*2, h = r5.h*2,
             red = r5.red, green = r5.green*.9, blue = r5.blue*.8,
             rotation = r2.rotation*5);

rectangle r7(x = r6.x - r6.w/2, y = r6.y - r6.h/2, w = r6.w*2, h = r6.h*2,
             red = r6.red, green = r6.green*.9, blue = r6.blue*.8,
             rotation = r2.rotation*6);
