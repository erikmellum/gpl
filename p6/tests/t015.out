gpl.cpp::main()
  input file(tests/t015.gpl)
  random seed(42)
  read_keypresses_from_standard_input(false)
  dump_pixels(true, file = results/t015.mypixels)
  symbol_table(true)
  print_symbol_table(true)
  graphics(true)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().


Printing the symbol table from main()
game_object r1
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(1),
    drawing_order = int(0),
    green = double(1),
    h = int(10),
    proximity = int(4),
    red = double(0.5),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(300),
    y = int(300)
  }
game_object r2
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.8),
    drawing_order = int(0),
    green = double(0.9),
    h = int(20),
    proximity = int(4),
    red = double(0.5),
    rotation = double(30),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(20),
    x = int(295),
    y = int(295)
  }
game_object r3
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.64),
    drawing_order = int(0),
    green = double(0.81),
    h = int(40),
    proximity = int(4),
    red = double(0.5),
    rotation = double(60),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(40),
    x = int(285),
    y = int(285)
  }
game_object r4
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.512),
    drawing_order = int(0),
    green = double(0.729),
    h = int(80),
    proximity = int(4),
    red = double(0.5),
    rotation = double(90),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(80),
    x = int(265),
    y = int(265)
  }
game_object r5
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.4096),
    drawing_order = int(0),
    green = double(0.6561),
    h = int(160),
    proximity = int(4),
    red = double(0.5),
    rotation = double(120),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(160),
    x = int(225),
    y = int(225)
  }
game_object r6
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.32768),
    drawing_order = int(0),
    green = double(0.59049),
    h = int(320),
    proximity = int(4),
    red = double(0.5),
    rotation = double(150),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(320),
    x = int(145),
    y = int(145)
  }
game_object r7
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.262144),
    drawing_order = int(0),
    green = double(0.531441),
    h = int(640),
    proximity = int(4),
    red = double(0.5),
    rotation = double(180),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(640),
    x = int(-15),
    y = int(-15)
  }
int window_height 600
string window_title "p6 t015.gpl -- compare to t015.jpg"
int window_width 600
gpl.cpp::main() Calling window->initialize().
gpl.cpp::main() Passing control to window->main_loop().

Printing the symbol table from signal_handler()
game_object r1
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(1),
    drawing_order = int(0),
    green = double(1),
    h = int(10),
    proximity = int(4),
    red = double(0.5),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(300),
    y = int(300)
  }
game_object r2
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.8),
    drawing_order = int(0),
    green = double(0.9),
    h = int(20),
    proximity = int(4),
    red = double(0.5),
    rotation = double(30),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(20),
    x = int(295),
    y = int(295)
  }
game_object r3
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.64),
    drawing_order = int(0),
    green = double(0.81),
    h = int(40),
    proximity = int(4),
    red = double(0.5),
    rotation = double(60),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(40),
    x = int(285),
    y = int(285)
  }
game_object r4
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.512),
    drawing_order = int(0),
    green = double(0.729),
    h = int(80),
    proximity = int(4),
    red = double(0.5),
    rotation = double(90),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(80),
    x = int(265),
    y = int(265)
  }
game_object r5
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.4096),
    drawing_order = int(0),
    green = double(0.6561),
    h = int(160),
    proximity = int(4),
    red = double(0.5),
    rotation = double(120),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(160),
    x = int(225),
    y = int(225)
  }
game_object r6
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.32768),
    drawing_order = int(0),
    green = double(0.59049),
    h = int(320),
    proximity = int(4),
    red = double(0.5),
    rotation = double(150),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(320),
    x = int(145),
    y = int(145)
  }
game_object r7
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0.262144),
    drawing_order = int(0),
    green = double(0.531441),
    h = int(640),
    proximity = int(4),
    red = double(0.5),
    rotation = double(180),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(640),
    x = int(-15),
    y = int(-15)
  }
int window_height 600
string window_title "p6 t015.gpl -- compare to t015.jpg"
int window_width 600
